Simplify detection and support of ansi codes

dmerejkowsky opened the pull request:
* Always call colorama.init()
* Implement CLICOLORS standard
* Remove the `mintty` workaround
* Remove workaround when setting terminal title

Note: this means that getting garbage output is less likely, but
it also means you now *have* to set CLICOLRS_FORCE=1 when
using programs that depends on cli-ui on mintty

Fix #51

dmerejkowsky committed:
Simplify detection and support of ansi codes

* Always call colorama.init()
* Implement CLICOLORS standard
* Remove the `mintty` workaround
* Remove workaround when setting terminal title

Note: this means that getting garbage output is less likely, but
it also means you now *have* to set CLICOLRS_FORCE=1 when
using programs that depends on cli-ui on mintty

dmerejkowsky committed:
documentation index: remove outdated stuff

The list of dependencies and the supported Python versions
are available on pypi.org and are always
up to date, contrary to this page in the doc.

blastrock commented on cli_ui/__init__.py:
```diff
@@ -75,6 +71,13 @@ def _setup(**kwargs: ConfigValue) -> None:
         CONFIG[key] = value
 
 
+def _init_coloram() -> None:
```
colorama?

dmerejkowsky replied:
yeah - windows works really fine until the keyboard bl

blastrock replied:
I mean there's a typo. Let me try the new suggestion feature of github.

```suggestion
def _init_colorama() -> None:
```

blastrock replied:
Oh you fixed it :)

blastrock commented on cli_ui/__init__.py:
```diff
@@ -145,43 +148,36 @@ def tuple(self) -> Tuple[Token, ...]:
+
+def write_title_string(message: str, fileobj: FileObj) -> None:
+    to_write = "\x1b]0;" + message + "\x07"
+    print("writing", repr(to_write))
```
is that debug?

dmerejkowsky replied:
oups

blastrock reviewed (changes requested):
I find the API strange, what if you have only one variable CLICOLORS_FORCE? If it's not set, guess with isatty, if it's set to 1, force colors, if it's set to 0 disable colors.

blastrock commented on cli_ui/__init__.py:
```diff
@@ -145,43 +148,35 @@ def tuple(self) -> Tuple[Token, ...]:
+    if os.environ.get("CLICOLORS") == "0":
+        return False
+    if os.environ.get("CLICOLORS_FORCE", "0") != "0":
         return True
```
According to the standard, if CLICOLORS_FORCE is enabled, you must enable colors. This condition should be checked first.

dmerejkowsky committed:
cli_ui: rename config_colors -> colors_enabled

dmerejkowsky committed:
info_progress: use write_and_flush

Reduces duplication

dmerejkowsky committed:
info_progress: use colors_enabled() helper

This fixes a bug where progress bar would not show up in some corner
cases (like using mintty with CLICOLORS_FORCE=1)

dmerejkowsky committed:
Rewrite main demo

* Don't monkey patch
* Sleep only when using info_progress()
* Avoid prompting the user
